src/listen/websocket.rs

```diff
@@ -461,15 +462,17 @@ macro_rules! send_message {
 async fn run_worker(
     ws_stream: WebSocketStream<MaybeTlsStream<tokio::net::TcpStream>>,
     mut message_tx: Sender<WsMessage>,
-    mut message_rx: Receiver<WsMessage>,
+    message_rx: Receiver<WsMessage>,
     mut response_tx: Sender<Result<StreamResponse>>,
     keep_alive: bool,
 ) -> Result<()> {
     // We use Vec<u8> for partial frames because we don't know if a fragment of a string is valid utf-8.
     let mut partial_frame: Vec<u8> = Vec::new();
-    let (mut ws_stream_send, mut ws_stream_recv) = ws_stream.split();
+    let (mut ws_stream_send, ws_stream_recv) = ws_stream.split();
+    let mut ws_stream_recv = ws_stream_recv.fuse();
     let mut is_open: bool = true;
     let mut last_sent_message = tokio::time::Instant::now();
+    let mut message_rx = message_rx.fuse();
```

Comment:
It won't let me comment on the `tokio::time::sleep_until` variable a few lines down.

I presume this doesn't need to be fused outside the loop, since the variable is created within the loop, so it isn't subject to the same polling issues as the other futures, which are called repeatedly within the loop?

Reply:
Also, since `message_rx` is fused here, shouldn't we *not* fuse it within the loop?

It [seems like](https://github.com/rust-lang/rust-clippy/issues/11143) `object.next().fuse()` can override the `object.fuse()` call and still cause rough behavior.

Reply:
Correct about sleep.  Each new iteration of the loop gets a new future, so any given sleep future will not get polled after the single `select_biased!` construct completes.

Reply:
Good catch on the `message_rx.next().fuse()`.  Fixed in 35b2083